src/Microsoft.Health.Dicom.SqlServer/Features/Store/SqlIndexDataStoreV35.cs

```diff
@@ -62,7 +63,7 @@ public override async Task<IReadOnlyList<InstanceMetadata>> RetrieveDeletedInsta
                                 seriesInstanceUid,
                                 sopInstanceUid,
                                 watermark,
-                                partitionKey),
+                                new Partition(partitionKey, Partition.UnknownName)),
                             new InstanceProperties()
```

Comment:
Same here, Defaulting to uknown is wrong. Either populate the name or remove the files